### ISSUE TITLE
add list a stock synonyms col to gbs download output format

### ISSUE THREAD

**Issue by bauchetg** [solgenomics/sgn#1193]
knowing the various names for some accession that will help!

**Comment by lukasmueller** [comment]
Also: allow to query all the synonyms for accessions based on a list

**Comment by dauglyon** [comment]
I think creating a general use function which takes a list of `stock.stock_id` `stock.uniquename` or (`stockprop.value`|`stock.uniquename`), and returns a hashmap of names like:
```javascript
{
    stock_id1: ["uniquename1","snyname1","snyname1"],
    stock_id2: ["uniquename2","snyname2","snyname2"],
}
```
That seems like it might be useful in multiple situations.
So something like
```perl
sub get_synonyms {
    my $self = shift;
    my $type = shift; # 'stock_id' | 'uniquename' | 'some_name'
    my @to_get = @{+shift};
    if ($type=='stock_id'){
        # ... query for matching names ...
    } elsif ($type=='uniquename'){
        # ... query for matching names ...
    } elsif ($type=='some_name'){
        # ... query for matching names ...
    } else {
        die;
    }
    # ...
    return %names;
}
```

**Comment by dauglyon** [comment]
*Something like this might already exist, but I have not found it.

**Comment by dauglyon** [comment]
Also add: stock list Unique-ifyer button, synonym search

**Comment by dauglyon** [comment]
another possible relevant ticket: #1244 #

**Comment by dauglyon** [comment]
@bauchetg how does this look?:

```
# Downloaded from SGN: Fri Sep  1 12:21:23 2017
# Protocol Id=1, Accession List: , Accession Ids: 38878,38879,38880,38881,38882,38883, Trial Ids: 139
# Synonyms: UG120002=(another_synonym) UG120004=(synonym_UG120004, syn2_UG120004, syn1_UG120004) UG120006=(UG120006_synonym)
Marker	UG120004|78264	UG120002|78313	UG120001|78341	UG120006|78351	UG120005|78398	UG120003|78418	UG120006|79773	UG120004|79798	UG120002|79831	UG120005|79953	UG120003|79982	UG120001|79997	
S5_36739	0	1	1	0	0	1	0	0	1	0	1	1
S13_92567	1	0	1	0	1	0	0	1	0	1	0	1
S69_57277	0	2	1	0	1	2	0	0	2	1	2	1
```

**Comment by bauchetg** [comment]
@dauglyon 
Nice however it may be a bit confusing here:
UG120002=(another_synonym): means no synonym?
UG120004=(synonym_UG120004, syn2_UG120004, syn1_UG120004) : how do synonyms differ?

Alternatively: I understand that it may not be easy to source and align synonyms (when existing..) with each columns, for simplicity  maybe we can  just rely on a list based tool we discussed that allow users to pull all existing synonyms  available.
Having this info would probably be enough for a user to match names.

**Comment by dauglyon** [comment]
@bauchetg `another_synonym` is a test synonym for `UG120002`.  `synonym_UG120004` `syn2_UG120004` and `syn1_UG120004` are all synonyms for `UG120004`. If there is not synonym for an accession it is not shown in that header row, if there are no synonyms for any accession, the synonyms header row is not present.

**Comment by dauglyon** [comment]
I am also currently working on an interface for viewing all synonyms in a list and optionally replacing them with unique names.

**Comment by bauchetg** [comment]
ok I see, I did not get it was a test output, in that case that should be just fine! Thanks for the clarification! Yep the interface will complete it.

**Comment by dauglyon** [comment]
@bauchetg I still need to add an option to overwrite the current list with the desynonymized version and an option to save as a new list but:
![2017-09-01 16_10_47](https://user-images.githubusercontent.com/5115845/29986202-5e0ebc9c-8f30-11e7-8acb-c55abb3a111a.gif)

**Comment by dauglyon** [comment]
Implemented in pull request #1267

**Comment by lukasmueller** [comment]
Nice!
